homeassistant/components/template/config.py

```diff
@@ -122,9 +122,11 @@ async def _async_resolve_blueprints(
             raise vol.Invalid("more than one platform defined per blueprint")
         if len(platforms) == 1:
             platform = platforms.pop()
-            for prop in (CONF_NAME, CONF_UNIQUE_ID, CONF_VARIABLES):
+            for prop in (CONF_NAME, CONF_UNIQUE_ID):
                 if prop in config:
                     config[platform][prop] = config.pop(prop)
+            if CONF_TRIGGER not in config and CONF_VARIABLES in config:
+                config[platform][CONF_VARIABLES] = config.pop(CONF_VARIABLES)
```

Comment:
Why is this changed? Maybe add a comment explaining what's going on?

Reply:
This PR allows variables to be executed between trigger and actions for trigger based template entities.  This means that the variables key cannot be removed when translating the blueprint configuration to the template configuration.

In theory, I suppose we could remove the if check all together and just let variables reside in the non-trigger-based template entity configurations.  I'm good with either path.

Reply:
> In theory, I suppose we could remove the if check all together and just let variables reside in the non-trigger-based template entity configurations.

Can you explain in some more detail what you mean?